tracer/src/Datadog.Trace.SourceGenerators/PublicApi/PublicApiGenerator.cs

```diff
@@ -0,0 +1,273 @@
+// <copyright file="PublicApiGenerator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Datadog.Trace.SourceGenerators.Helpers;
+using Datadog.Trace.SourceGenerators.PublicApi.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Datadog.Trace.SourceGenerators.PublicApi;
+
+/// <inheritdoc />
```

Comment:
Not really, no 😄 It inherits the docs from `IIncrementalGenerator` I think, but mostly it stops stylecop complaining about documentation of a class that noone can ever touch 😉

Reply:
If we ever generate (html) docs from the xml comments, it copies the xml from the parent.

> stops stylecop complaining about documentation

@andrewlock: one way to fix this is to make the class `internal`, but then you would need to figure out the `[InternalsVisibleTo]` so we can use this from `Datadog.Trace`. Shouldn't be hard since this assembly doesn't need to be strongly named.